pkg/controller/replicaset/replica_set_test.go

```diff
@@ -1982,6 +2005,99 @@ func TestGetPodsToDelete(t *testing.T) {
 	}
 }
 
+func withPodNameSuffix(pod *v1.Pod, suffix string) *v1.Pod {
+	newPod := pod.DeepCopy()
+	newPod.Name = fmt.Sprintf("%s-%s", pod.Name, suffix)
+	return newPod
+}
+
+func TestComparingSameNodeSpreadOnDeletion(t *testing.T) {
+	labelMap := map[string]string{"name": "foo"}
+	rs := newReplicaSet(6, labelMap)
+	timeADayAgo := time.Now().Add(-time.Hour * 24)
+
+	// scheduled, running, ready pods on fake-node-1
+	podAOnNode1 := newPod("pod-a-on-node-1", rs, v1.PodRunning, &metav1.Time{Time: timeADayAgo.Add(-time.Hour)}, true)
```

Comment:
Just to make sure you understood how the logarithmic comparisons work. The basic idea is that if a Pod has been running for less than half the time as another, it is sorted before. Then, in the test cases that you are setting, you are keeping the times close to each other so they have the same logarithm. Correct?